### ISSUE TITLE
Cannot watch config : stat : no such file or directory

### ISSUE THREAD

**Issue by peter-bartyik** [influxdata/telegraf#13272]
### Relevant telegraf.conf

```toml
[[outputs.influxdb_v2]]
   urls = ["http://influxdb:8086"]
   token = ""
   organization = "myorg"
   bucket = "mybucket"
```

### Logs from Telegraf

```text
root@telegraf:/# telegraf --watch-config notify --debug
2023-05-15T12:38:20Z I! Loading config: /etc/telegraf/telegraf.conf
2023-05-15T12:38:20Z W! Cannot watch config : stat : no such file or directory
2023-05-15T12:38:20Z I! Starting Telegraf 1.26.2
2023-05-15T12:38:20Z I! Available plugins: 235 inputs, 9 aggregators, 27 processors, 22 parsers, 57 outputs, 2 secret-stores
2023-05-15T12:38:20Z I! Loaded inputs: disk
2023-05-15T12:38:20Z I! Loaded aggregators:
2023-05-15T12:38:20Z I! Loaded processors:
2023-05-15T12:38:20Z I! Loaded secretstores:
2023-05-15T12:38:20Z I! Loaded outputs: influxdb_v2
2023-05-15T12:38:20Z I! Tags enabled: host=telegraf
2023-05-15T12:38:20Z I! [agent] Config: Interval:10s, Quiet:false, Hostname:"telegraf", Flush Interval:10s
2023-05-15T12:38:20Z D! [agent] Initializing plugins
2023-05-15T12:38:20Z D! [agent] Connecting outputs
2023-05-15T12:38:20Z D! [agent] Attempting connection to [outputs.influxdb_v2]
2023-05-15T12:38:20Z D! [agent] Successfully connected to outputs.influxdb_v2
2023-05-15T12:38:20Z D! [agent] Starting service inputs
2023-05-15T12:38:30Z E! [outputs.influxdb_v2] When writing to [http://influxdb:8086]: failed to write metric to mybucket (401 Unauthorized): unauthorized: unauthorized access
2023-05-15T12:38:30Z D! [outputs.influxdb_v2] Buffer fullness: 6 / 10000 metrics
2023-05-15T12:38:30Z E! [agent] Error writing to outputs.influxdb_v2: failed to send metrics to any configured server(s)
```

### System info

Telegraf 1.26.2 (git: HEAD@a3a884a1), 6.2.1-arch1-1, Docker version 23.0.6, build ef23cbc431, Docker Compose version 2.17.3

### Docker

version: "3.7"
services:
  grafana:
    image: grafana/grafana-oss:latest
    container_name: grafana
    hostname: grafana
    restart: always
    ports:
      - 3000:3000
    depends_on:
      - influxdb
    volumes:
      - grafana-data:/var/lib/grafana
      - ./grafana/provisioning/datasources:/etc/grafana/provisioning/datasources
  telegraf:
    image: telegraf:latest
    container_name: telegraf
    hostname: telegraf
    restart: always
    depends_on:
      - influxdb
    volumes:
      - ./telegraf/telegraf.conf:/etc/telegraf/telegraf.conf
    ports:
      - 8125:8125
      - 8092:8092
      - 8094:8094
    command: telegraf --watch-config notify --debug
  influxdb:
    image: influxdb:latest
    container_name: influxdb
    hostname: influxdb
    restart: always
    ports:
      - 8086:8086
    volumes:
      - influx-data:/var/lib/influxdb2

volumes:
  influx-data:
  grafana-data:

### Steps to reproduce

1. do not set up influxdb, so telegraf can't connect to it
2. docker-compose up
3. change config file from port 8086 to 8085
...

### Expected behavior

Telegraf should not produce the error message `W! Cannot watch config : stat : no such file or directory`
When changing telegraf.conf port for influxdb from 8085 from 8086 telegraf should reload config and attempt to connect on port 8085 instead of 8086

### Actual behavior

Telegraf produces error message `W! Cannot watch config : stat : no such file or directory`
Telegraf does not reload config when telegraf.conf is changed, keeps attempting to connect on port 8086

### Additional info

I'm testing with the influxdb output config parameters because I've not yet set up any meaningful input that I could test with.
When testing within the container, the issue is the same. The command `stat` is available:
```
[user@localhost ~ ]$ docker exec -u telegraf telegraf which stat
/usr/bin/stat
[user@localhost ~ ]$
```

The default config is active, I've only enabled outputs.influxdb_v2 with the above specified changes.

I'd have also liked to ask for advice on this on slack, however the invite link appears to be expired.

**Comment by peter-bartyik** [comment]
Yep, that works. Can the `--help` message and documentation be updated to reflect the behaviour?

**Comment by powersj** [comment]
@peter-bartyik yep! can you take a look at the change in #13285 and let me know if that is enough :)

**Comment by peter-bartyik** [comment]
looks good, thanks for the quick action!
